staging/src/k8s.io/client-go/discovery/content_type.go

```diff
@@ -0,0 +1,124 @@
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"mime"
+	"reflect"
+	"sort"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	AcceptV1            = runtime.ContentTypeJSON
+	AcceptV2Beta1Params = sortedContentTypeParams{
+		params: map[string]string{
+			"g":  "apidiscovery.k8s.io",
+			"v":  "v2beta1",
+			"as": "APIGroupDiscoveryList",
+		},
+	}
+	// Aggregated discovery content-type (v2beta1). NOTE: content-type parameters
+	// MUST be ordered (g, v, as) for server in "Accept" header (BUT we are resilient
+	// to ordering when comparing returned values in "Content-Type" header).
+	AcceptV2Beta1 = runtime.ContentTypeJSON + ";" + AcceptV2Beta1Params.String()
+	// Prioritize aggregated discovery by placing first in the order of discovery accept types.
+	acceptDiscoveryFormats = AcceptV2Beta1 + "," + AcceptV1
+)
+
+// sortedContentTypeParams encapsulated content type parameters, where
+// the discovery content-type parameters will be ordered when printed,
+// but where parameter ordering will not matter when comparing.
+type sortedContentTypeParams struct {
+	params map[string]string
+}
+
+// sortedKeys returns content-type parameters sorted so keys
+// "g" > "v" > "as". No deterministic ordering if keys are not
+// discovery content-type parameters {"g", "v", "as"}.
+func (p *sortedContentTypeParams) sortedKeys() []string {
+	keys := make([]string, len(p.params))
+	for key := range p.params {
+		keys = append(keys, key)
+	}
+	// Ordering: "g" > "v" > "as"
+	sort.Slice(keys, func(i, j int) bool {
+		if strings.EqualFold(keys[i], "g") {
+			return true
+		} else if strings.EqualFold(keys[i],"as") {
+			return false
+		} else if strings.EqualFold(keys[i], "v") && strings.EqualFold(keys[j], "as") {
+			return true
+		} else {
+			return false
+		}
+	})
+	return keys
+}
+
+// String prints the content-type params using the sortedKeys array to
+// print the params in the order of the "sortedKeys" field.
+func (p *sortedContentTypeParams) String() string {
+	keyValues := []string{}
+	for _, key := range p.sortedKeys() {
+		if value, ok := p.params[key]; ok {
+			keyValues = append(keyValues, key+"="+value)
+		}
+	}
+	return strings.Join(keyValues, ";")
+}
+
+// Equals returns true if the passed parameters map is the same as the
+// receivers parameters. NOTE: the ordering of the parametes does NOT matter.
+func (p *sortedContentTypeParams) Equals(params map[string]string) bool {
+	return reflect.DeepEqual(p.params, params)
+}
+
+// IsV1ContentType returns true if the passed content-type string is
+// discovery version 1 (e.g. "application/json"). NOTE: The content-type
+// parameters can be anything OTHER than v2beta1 parameters.
+// Examples:
+//
+//	"application/json" = true
+//	"application/json; charset=UTF-8" = true
+//	"application/json; g=apidiscovery.k8s.io;v=v2beta1;as=APIGroupDiscoveryList" = false
+func IsV1ContentType(contentType string) bool {
+	base, params, err := mime.ParseMediaType(contentType)
```

Comment:
avoid calling ParseMediaType twice (once in IsV1ContentType, once in IsV2Beta1ContentType), that can be relatively expensive